### PR TITLE
Log error and not create ndj file when wos extract error occurs.

### LINKED ISSUE #285: WOS client should log non-200 responses
It currently does not check the response code or log anything. As a result, the user has no way of knowing if there was an error.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #285

### CODE REVIEW

File: lib/rialto/etl/cli/composite_etl.rb [hunk 1]
```diff
@@ -106,18 +107,27 @@ def file_prefix
...
           if File.exist?(extract_file) && !force
             say "file #{extract_file} already exists, skipping. use -f to force overwrite"
             return extract_file
           end
-          results = perform_extract(row)
+          # Don't write file if an exception occurs
+          begin
+            results = perform_extract(row)
+          rescue StandardError
+            say "Skipping #{extract_file} because an error occurred while extracting"
```
Comment (mjgiarlo) [inline lib/rialto/etl/cli/composite_etl.rb]: What would you think about surfacing the exception class and message here too? Useful, or not so much?

File: lib/rialto/etl/cli/composite_etl.rb [hunk 2]
```diff
@@ -106,18 +107,27 @@ def file_prefix
...
+            return extract_file
```
Comment (jcoyne) [inline lib/rialto/etl/cli/composite_etl.rb]: We're returning a name to a non-existent file?

File: lib/rialto/etl/service_client/web_of_science_client.rb [hunk 3]
```diff
@@ -56,10 +56,12 @@ def perform_initial_query!
...
         def connect_with_retries(path:)
           response = client.get(path)
+          raise "#{response.reason_phrase}: #{response.status}  (#{response.body})" unless response.success?
           JSON.parse(response.body)
         rescue StandardError => exception
-          logger.warn "Error fetching #{path}: #{exception.message} (#{exception.class})"
-          {}
+          msg = "Error fetching #{path}: #{exception.message} (#{exception.class})"
+          logger.warn msg
+          raise msg
```
Comment (jcoyne) [inline lib/rialto/etl/service_client/web_of_science_client.rb]: Why don't we re-raise the original error, so that it's easier to track down the cause?

Comment (mjgiarlo) [inline lib/rialto/etl/service_client/web_of_science_client.rb]: Does this conflict with what's in the `RetriableConnectionFactory`? https://github.com/sul-dlss/rialto-etl/blob/master/lib/rialto/etl/service_client/retriable_connection_factory.rb#L30-L55 We want to retry the errors and errors conditions that are specified there and not bomb out the whole run.

File: lib/rialto/etl/cli/composite_etl.rb (hunk 2 above)
Comment (justinlittman) [reply lib/rialto/etl/cli/composite_etl.rb]: Yes, the code that invokes this checks to see if the file exists and is not empty.

An empty file is a normal result of this method, as it indicates that there are no publications.

File: lib/rialto/etl/service_client/web_of_science_client.rb (hunk 3 above)
Comment (justinlittman) [reply lib/rialto/etl/service_client/web_of_science_client.rb]: I changed this so re-raises the original error.

File: lib/rialto/etl/cli/composite_etl.rb (hunk 1 above)
Comment (justinlittman) [reply lib/rialto/etl/cli/composite_etl.rb]: Done.

**Comment by justinlittman** [comment]
I'm going to change one of the tests, so hold off on further review.

**Comment by justinlittman** [comment]
Made my fixes. Now more directly tests handling of errors thrown by extractor in composite ETL.

**Pull request merged**